### PR TITLE
[12.0][ENH] l10n_th_vendor_tax_invoice to apply with tax sequence

### PR DESCRIPTION
**Pull request by kittiu**
Allow using same tax sequence for both tax on invoice and payment.

### CODE REVIEW

**Review by ps-tubtim — APPROVED** [review]
Code LGTM

File: e -aml.balance as tax_amount [hunk 1]
```diff
@@ -0,0 +1,236 @@
+
```
Comment (newtratip) [inline e -aml.balance as tax_amount]: What is this ?

File: e_amount [hunk 2]
```diff
@@ -0,0 +1,236 @@
+
```
Comment (newtratip) [inline e_amount]: What is this ?

File: tate = posted [hunk 3]
```diff
@@ -0,0 +1,236 @@
+
```
Comment (newtratip) [inline tate = posted]: What is this ?

File: l10n_th_vat_report/reports/vat_report.py [hunk 4]
```diff
@@ -46,21 +46,27 @@ class VatReport(models.TransientModel):
...
+            SELECT id, company_id, name, account_id, tax_invoice, partner_id,
+                date, tax_date, tax_amount,
+                CASE when tax_amount < 0 then -tax_base_amount
+                else tax_base_amount end as tax_base_amount
+            FROM (
+                SELECT aml.id as id, am.company_id, am.name, aml.account_id,
+                    aml.tax_invoice, aml.partner_id, aml.date, aml.tax_date,
+                    abs(aml.tax_base_amount) as tax_base_amount,
+                    CASE WHEN tx.type_tax_use = 'sale'
+                    then -aml.balance
```
Comment (Saran440) [inline l10n_th_vat_report/reports/vat_report.py]:
```suggestion
                    THEN -aml.balance
                    WHEN tx.type_tax_use = 'adjustment' THEN abs(aml.balance)
```

Supported Tax adjustment

Comment (kittiu) [reply l10n_th_vat_report/reports/vat_report.py]: Done.

Comment (kittiu) [reply l10n_th_vat_report/reports/vat_report.py]: @Saran440 can I merge?

**Review by Saran440 — APPROVED** [review]
LGTM :+1:

**Pull request merged**